pkg/backend/httpstate/journal/snapshot_test.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package journal
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/httpstate/client"
+	"github.com/pulumi/pulumi/pkg/v3/engine"
+	"github.com/pulumi/pulumi/pkg/v3/secrets/b64"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var updateIdentifier = client.UpdateIdentifier{
+	StackIdentifier: client.StackIdentifier{
+		Owner:   "org",
+		Project: "project",
+		Stack:   tokens.MustParseStackName("stack"),
+	},
+	UpdateKind: apitype.UpdateUpdate,
+	UpdateID:   "update-id",
+}
+
+func startTestServer(t *testing.T, handler func(w http.ResponseWriter, body apitype.JournalEntries)) *client.Client {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		raw, err := gzip.NewReader(req.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		var body apitype.JournalEntries
+		if err := json.NewDecoder(raw).Decode(&body); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		handler(w, body)
+	}))
+	t.Cleanup(srv.Close)
+
+	sink := diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{
+		Color: colors.Never,
+	})
+	return client.NewClient(srv.URL, "test-token", true, sink)
```

Comment:
Note: this is fine because `client.NewClient` creates a new transport if we pass `insecure: true`.  (We've had issues with httptest servers using the `DefaultClient` before, because the httptest.NewServer closes all idle connections on a transport when it closes.